packages/frontend/src/lib/creatorReadHooks.ts

```diff
@@ -132,7 +140,123 @@ export const useClaimableAmount = ({ beneficiary }: { beneficiary?: string }) =>
   return {
     isLoading,
     claimableAmounts,
-    claimableAmount: contractsChainId && claimableAmounts[contractsChainId],
+    ...(contractsChainId ? { claimableAmount: claimableAmounts[contractsChainId] } : {}),
+    contractChain: contractsChain,
+    contractChainId: contractsChainId,
+    refetch: async () => {
+      refetch(contractsChainId)
+    },
+  }
+}
+
+/**
+ * Returns the pool address for a given creator
+ */
+export const usePoolAddress = ({ beneficiary }: { beneficiary?: string }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const { contractsChain, contractsChainId, contracts } = useDeployments()
+  const [poolAddresses, setPoolAddresses] = useState<{
+    [_: string]: string | false
+  }>({})
+
+  const refetch = async (chainId: number | undefined) => {
+    if (!beneficiary || !contracts || !chainId) return
+    setIsLoading(true)
+    setPoolAddresses((prev) => ({
+      ...prev,
+      [chainId]: undefined,
+    }))
+    let poolAddress: string | false
+    try {
+      const provider = getDefaultProvider(env.rpcUrls[chainId as keyof typeof env.rpcUrls])
```

Comment:
I see this rather complex expression being used multiple times. Could it be extracted into a function `getProvider` in `@lib`?

Reply:
Did that in https://github.com/yieldgate/yieldgate/pull/56/commits/7c6b5af45f7e8ade32862d84f12307607b8ee129.

Actually I would love to use the wagmi `getProvider` function for this, but they are currently only exposing the hook (and I had problems with this if I remember correctly). I opened an issue there as well: https://github.com/wagmi-dev/wagmi/issues/901

Reply:
beautiful ✨